### PR TITLE
feat(app): create parameters ODD screen

### PR DESCRIPTION
**Pull request by jerader**
closes AUTH-116

# Overview

This PR creates the Parameters ODD screen and leaves the buttons to wire up later

# Test Plan

Send a protocol to the ODD and click on the "start setup" button and you should see the parameters page. The buttons are not wired up yet. If you turn the feature flag off, you should see the setup as usual.

# Changelog

- create `ProtocolSetupParameters` and test
- plug in the component to `ProtocolSetup`
- create new hook `useProtocolHasRunTimeParameters` which just returns a boolean
- create test fixture for run time parameters key
- modify `ProtocolSetupStep` to include an optional description

# Review requests

see test plan

# Risk assessment

low

### CODE REVIEW

File: app/src/pages/ProtocolSetup/__tests__/ProtocolSetup.test.tsx [hunk 1]
```diff
@@ -72,6 +74,7 @@ Object.defineProperty(window, 'IntersectionObserver', {
 
 let mockHistoryPush = vi.fn()
 
+vi.mock('../../ProtocolDetails/fixtures')
```
Comment (koji) [inline app/src/pages/ProtocolSetup/__tests__/ProtocolSetup.test.tsx]: need to mock fixtures?

Comment (jerader) [reply app/src/pages/ProtocolSetup/__tests__/ProtocolSetup.test.tsx]: lol i have no idea why i did that, fixing it now!

File: app/src/organisms/ProtocolSetupParameters/index.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+          >
+            {t('restore_default')}
+          </StyledText>
+          <SmallButton
+            buttonCategory="rounded"
+            buttonText={t('confirm_values')}
+            onClick={handleConfirmValues}
+          />
+        </Flex>
+      </Flex>
```
Comment (koji) [inline app/src/organisms/ProtocolSetupParameters/index.tsx]: Can we use Child Navigation component for the header?

https://s3-us-west-2.amazonaws.com/opentrons-components/edge/index.html?path=/story/odd-organisms-childnavigation--title-with-two-buttons

Comment (jerader) [reply app/src/organisms/ProtocolSetupParameters/index.tsx]: oops yes, for sure. I didn't realize we have a component for this. thanks!

File: app/src/organisms/ProtocolSetupParameters/index.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        onClickButton={handleConfirmValues}
+        buttonText={t('confirm_values')}
+        secondaryButtonProps={{
+          buttonType: 'tertiaryLowLight',
+          buttonText: t('restore_default'),
+          onClick: () => console.log('TODO: wire this up!'),
+        }}
+      />
+      <Flex
+        alignItems={ALIGN_CENTER}
```
Comment (koji) [inline app/src/organisms/ProtocolSetupParameters/index.tsx:218]: Sorry I forgot to mention this.
The ODD cannot show the first item properly without marginTop.

```suggestion
       marginTop="7.75rem"
        alignItems={ALIGN_CENTER}
```
without marginTop
The first item is cut off.

with marginTop (I tested the suggestion on ODD)
Screenshot 2024-03-18 at 12 37 31 PM

File: app/src/organisms/ProtocolSetupParameters/index.tsx [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                    choice => choice.value === defaultValue
+                  )
+                  if (choice != null) {
+                    return choice.displayName
+                  }
+                }
+                break
+            }
+            return ''
+          }
```
Comment (koji) [inline app/src/organisms/ProtocolSetupParameters/index.tsx]: can we move this function outside of map?

**Comment by koji** [comment]
Screenshot 2024-03-18 at 3 42 00 PM

seems that the font size of `value` is a little bit smaller than design.

the design 
font size: 28px
font height: 36px 

Could you check that the font config aligns with the design.

File: app/src/organisms/ProtocolSetupParameters/__tests__/ProtocolSetupParameters.test.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  let props: React.ComponentProps<typeof ProtocolSetupParameters>
+
+  beforeEach(() => {
+    props = {
+      runId: 'mockId',
+      setSetupScreen: vi.fn(),
+    }
+    vi.mocked(useMostRecentCompletedAnalysis).mockReturnValue({
+      runTimeParameters: mockRunTimeParameterData,
+    } as any)
```
Comment (koji) [inline app/src/organisms/ProtocolSetupParameters/__tests__/ProtocolSetupParameters.test.tsx]: Probably using vitest-when would be better.

```suggestion
import { when } from 'vitest-when'

const RIN_ID = 'mockId'
    when(vi.mocked(useMostRecentCompletedAnalysis)).calledWith(RUN_ID).thenReturn({
      runTimeParameters: mockRunTimeParameterData,
    } as any)
```

File: app/src/pages/ProtocolDetails/fixtures.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
+import type { RunTimeParameter } from '@opentrons/shared-data'
+
+export const mockRunTimeParameterData: RunTimeParameter[] = [
+  {
+    displayName: 'Dry Run',
+    variableName: 'DRYRUN',
+    description: 'a dry run description',
+    type: 'boolean',
+    default: false,
```
Comment (y3rsh) [inline app/src/pages/ProtocolDetails/fixtures.ts:9]: I left out the `value` key in my examples by mistake.  All these should have a value key that from the desktop app perspective will always have the same value as the default for MVP.

**Comment by jerader** [comment]
> seems that the font size of `value` is a little bit smaller than design.
> 
> the design font size: 28px font height: 36px
> 
> Could you check that the font config aligns with the design.

@koji the design comments make me think we might change the size of the font down to 22px, is it ok if i leave it as is for now? i think it makes sense to just merge this PR in (after i address your other comments) and come back later during DQA since the widths of the subtitles aren't finalized anyway.

**Comment by koji** [comment]
> > seems that the font size of `value` is a little bit smaller than design.
> > the design font size: 28px font height: 36px
> > Could you check that the font config aligns with the design.
> 
> @koji the design comments make me think we might change the size of the font down to 22px, is it ok if i leave it as is for now? i think it makes sense to just merge this PR in (after i address your other comments) and come back later during DQA since the widths of the subtitles aren't finalized anyway.

@jerader yes. I think we just need a new ticket or use an existing ticket to track the issue.

**Comment by jerader** [comment]
> @jerader yes. I think we just need a new ticket or use an existing ticket to track the issue.

@koji made a ticket: https://opentrons.atlassian.net/browse/AUTH-215

**Review by koji — APPROVED** [review]
lgtm
once resolve the conflict, this pr is good to merge.

**Pull request merged**